pkg/app/piped/cmd/piped/piped.go

```diff
@@ -470,18 +473,38 @@ func (p *piped) createAPIClient(ctx context.Context, address, projectID, pipedID
 }
 
 // loadConfig reads the Piped configuration data from the specified file.
-func (p *piped) loadConfig() (*config.PipedSpec, error) {
-	cfg, err := config.LoadFromYAML(p.configFile)
-	if err != nil {
-		return nil, err
+func (p *piped) loadConfig(ctx context.Context) (*config.PipedSpec, error) {
+	extract := func(cfg *config.Config) (*config.PipedSpec, error) {
+		if cfg.Kind != config.KindPiped {
+			return nil, fmt.Errorf("wrong configuration kind for piped: %v", cfg.Kind)
+		}
+		if p.enableDefaultKubernetesCloudProvider {
+			cfg.PipedSpec.EnableDefaultKubernetesCloudProvider()
+		}
+		return cfg.PipedSpec, nil
 	}
-	if cfg.Kind != config.KindPiped {
-		return nil, fmt.Errorf("wrong configuration kind for piped: %v", cfg.Kind)
+
```

Comment:
Just in case, why not check if both given like: `p.configFile != "" && p.configInGCPSecretManager != ""`

Reply:
👍